stix_shifter_modules/trustar/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,252 @@
+from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import ObservationExpression, ComparisonExpression, \
+    ComparisonExpressionOperators, ComparisonComparators, Pattern, \
+    CombinedComparisonExpression, CombinedObservationExpression, ObservationOperators
+from stix_shifter_utils.stix_translation.src.utils.transformers import TimestampToMilliseconds
+from stix_shifter_utils.stix_translation.src.json_to_stix import observable
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from datetime import datetime
+import base64
+import json
+import logging
+import re
+
+# Source and destination reference mapping for ip and mac addresses.
+# Change the keys to match the data source fields. The value array indicates the possible data type that can come into from field.
+REFERENCE_DATA_TYPES = {"SourceIpV4": ["ipv4", "ipv4_cidr"],
+                        "SourceIpV6": ["ipv6"],
+                        "DestinationIpV4": ["ipv4", "ipv4_cidr"],
+                        "DestinationIpV6": ["ipv6"]}
+
+logger = logging.getLogger(__name__)
+
+
+class QueryStringPatternTranslator:
+    
+    return_obj = dict()
+
+    def __init__(self, pattern: Pattern, data_model_mapper):
+        self.dmm = data_model_mapper
+        self.comparator_lookup = self.dmm.map_comparator()
+        self.pattern = pattern
+        self.translated = self.parse_expression(pattern)
+
+    @staticmethod
+    def _format_set(values) -> str:
+        gen = values.element_iterator()
+        return "({})".format(' OR '.join([QueryStringPatternTranslator._escape_value(value) for value in gen]))
+
+    @staticmethod
+    def _format_match(value) -> str:
+        raw = QueryStringPatternTranslator._escape_value(value)
+        if raw[0] == "^":
+            raw = raw[1:]
+        else:
+            raw = ".*" + raw
+        if raw[-1] == "$":
+            raw = raw[0:-1]
+        else:
+            raw = raw + ".*"
+        return "{}".format(raw)
+
+    @staticmethod
+    def _format_equality(value) -> str:
+        return "{}".format(value)
+
+    @staticmethod
+    def _format_like(value) -> str:
+        value = "{value}".format(value=value)
+        return QueryStringPatternTranslator._escape_value(value)
```

Comment:
are you sure the LIKE operator translation is correct? translating this pattern `[ipv4-addr:value LIKE '127.0.0.%']` returns `"{\"searchTerm\": \"127.0.0. \", \"from\": 1656633600000, \"to\": 1658880300000}"`

the search term seems wrong!

Reply:
Done the changes now this query is giving  "{\"searchTerm\": \"127.0.0.%\", \"from\": 1656633600000, \"to\": 1658880300000}"
<img width="1363" alt="Screenshot 2022-12-17 at 12 46 34 PM" src="https://user-images.githubusercontent.com/98740866/208230501-c9cbca56-3061-4b5f-a4d2-8075ca102259.png">